tasks/LinearTemperatureSensorTask.cpp

```diff
@@ -0,0 +1,81 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "LinearTemperatureSensorTask.hpp"
+
+using namespace raw_io;
+using namespace base::samples;
+
+LinearTemperatureSensorTask::LinearTemperatureSensorTask(std::string const& name)
+    : LinearTemperatureSensorTaskBase(name)
+{
+}
+
+LinearTemperatureSensorTask::~LinearTemperatureSensorTask()
+{
+}
+
+// Convert the voltage in temperature
+Temperature temperature(base::Time const& timestamp,
+    double voltage,
+    LinearTemperatureSensorSettings const& sensor_config)
+{
+    auto temperature =
+        sensor_config.min_temperature +
+        (voltage - sensor_config.min_voltage) *
+            ((sensor_config.max_temperature - sensor_config.min_temperature) *
+                (1 / (sensor_config.max_voltage - sensor_config.min_voltage)));
```

Comment:
It's all OK, but the more intuitive way to express proportionality to me is

`(x - min) / (max - min) * range;`